server/src/main/java/org/elasticsearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -261,7 +266,7 @@ public void tryAction(ActionListener<T> listener) {
 
             @Override
             public boolean shouldRetry(Exception e) {
-                return retriesSupported && retryableException(e);
+                return retriesSupported && clusterService.state().nodes().nodeExists(targetNode) && retryableException(e);
```

Comment:
This feels really dirty to me. I mainly opened this to illustrate where the problem lies, not sure this is an appropriate fix. Ideally, I'd hope we would have some mechanism for cancelling the retry if a node is dropped form the cluster but I couldn't find an easy way of building that (maybe someone else can?).

Reply:
`ClusterStateObserver` is the magical thing here. It should be registered whenever we schedule a retry, and early cancel a scheduled retry when the target node drops from the cluster.

Reply:
Why don't we just have a single cluster state observer for `RemoteRecoveryTargetHandler`? We already have the whole cancellation infrastructure already built. Adding individual ones seems unnecessary.

Reply:
I didn't look at it in detail. Either is fine by me. Looks like @tbrooks8 volunteered to be a reviewer :)